core/src/validator.rs

```diff
@@ -179,6 +180,13 @@ impl Validator {
         let exit = Arc::new(AtomicBool::new(false));
         let bank_info = &bank_forks_info[0];
         let bank = bank_forks[bank_info.bank_slot].clone();
+
+        // Only do this check if started as a bootstrap leader.
+        // Normal validators can encounter newly-created vote accounts, which to be synced with this new validator.
+        if entrypoint_info_option.is_none() && !config.voting_disabled {
```

Comment:
Normal validators where the ones getting tripped up with the lack for vote account, not the bootstrap leader or a blockstreamer node.   How about we the vote account check moves from `core/` into `validator/`, and the check happens over RPC to the entrypoint (so the check does not occur for a bootstrap leader, which should be fine)

Reply:
Hm, that sounds nice! I was shorted sighted...

Reply:
@mvines I've tried to implement what you proposed [like this](https://github.com/solana-labs/solana/pull/6790/files?utf8=%E2%9C%93&diff=split&w=1#diff-1f422d7e73618aca7bce9935356de012R208). Could you just take a light look at it? If this approach looks good to proceed, I'll polish this up further including rebaseing on now-approved preparatory PR #6812 (thx for approvals, btw!).

Also, I rather want to keep the vote account check for bootstrapping nodes in the `./core/`. As this may sound redundant/excessive, I think this is useful to new prospective contributors fiddling the `solana-validator` command, and I like the completeness of the check for both cases of bootstrapping/normal nodes

Reply:
I think I'd prefer to remove the check from here to avoid the semi-duplication.  The bootstrap node is a little special, it's only started by one person and they're really sure about it, so I don't mind the lack of vote check for the bootstrap node

Reply:
I see. Just restored `./core/src/validator.rs` to the original state. :smile: 

Now this PR touches only `./validator/src/main.rs`.
/me feels safe :tongue: